nomad/state/state_store.go

```diff
@@ -384,6 +396,9 @@ func (s *StateStore) DeleteJob(index uint64, jobID string) error {
 	if _, err = txn.DeleteAll("job_summary", "id", jobID); err != nil {
```

Comment:
@dadgar `Delete` needs the object to be deleted, `DeleteAll` will delete the row which matches the job id since we are using that as the index in the schema.